### DISCUSSION TITLE
Allow configuring a timeout

### DISCUSSION THREAD

**Discussion by maggie44** [mysticaltech/terraform-hcloud-kube-hetzner#1131] [Ideas]
### Description

Sometimes when there are issues with connectivity or on Hetzner API, the retries continue and it gets stuck in a loop:

```
module.kube-hetzner.hcloud_firewall.k3s: Still modifying... [id=1141391, 4m20s elapsed]
```

Ideally we would be able to set a max retries or a timeout to stop this happening. 

I have raised this as an issue instead of a discussion as it is quite problematic when using k-hetzner within continuous integration flows. I came back to find an earlier run had gobbled up all my build minutes 💵 .

**Comment by mysticaltech** [comment]
I feel your pain @maggie44. Please would you like to try submitting a PR for this?  **Otherwise please pin-point exactly where is the sticking point you are observing. Is it just for the firewall?**

Also please try terraform init -upgrade, I would guess the latest hetzner module handles retries automatically. At least I think I saw that in their release note at some point.

But PRs most welcome as always.

**Comment by maggie44** [comment]
It's not just the firewall module, I see here for example it continuing for 43 minutes: https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/868

It's a little tricker to come up with a good solution without knowing much about Terraform. In other similar systems where different calls are made to do different things we might set timeouts based on each request, maybe altered by expected time. Or just allowing a max time for the whole process to run. In GitHub Actions you can set timeouts which would apply to this too, but if it's possible I think it would be a good idea to have in the Terraform process too. 

I'm hoping the issue will open a bit of dialog, and can flush out some ideas and a good option. I suspect there are also others who have used Terraform elsewhere and may know of some precedent. I will keep thinking about it too.

**Comment by mysticaltech** [comment]
Hopefully this is now solved. Even though we do not allow configuring a timeout, we should have reached something stable, unless Hetzner has issues at that particular moment.
